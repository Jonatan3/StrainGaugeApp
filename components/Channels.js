import React from 'react'
import { StyleSheet, View, Text, FlatList, TouchableWithoutFeedback, Button } from 'react-native'
import { Dimensions } from 'react-native';
import StreamingConstructor from '../constructors/StreamingConstructor';
import QRCodeButton from './QRCodeButton';

export default function Channels({ navigation }) {
  const windowWidth = (parseInt(Dimensions.get('window').width) / 2) - 24

  return (
    <View style={{ flex: 1 }}>
      <FlatList
        data={StreamingConstructor.getDummyData()}
        style={{ height: '100%' }}
        keyExtractor={(item, index) => item.id}
        renderItem={({ item, index }) => (
          <TouchableWithoutFeedback onPress={() => navigation.push('Strain Gauge', {itemId: item.id})}>
            <View>
              <View style={[styles.item, {backgroundColor: index % 2 == 0  ? "#fff" : "#F8F8FD"}]}>
                <View style={{ flexDirection: 'row' }}>
                  <View style={{ flex: 1 }}>
                    <Text style={styles.name}> {item.name} </Text>
                  </View>
                  <View style={{ flex: 1 }}>
                    <Text style={styles.index}> {index < 9 ? '0' + (index + 1) : (index + 1) } </Text>
                  </View>
                </View>
                <Text> {item.type} </Text>
              </View>

            </View>
          </TouchableWithoutFeedback>
        )}
      />
      <View style={{ position: 'absolute', top: 630, right: 80 }}>
        <QRCodeButton onPressHandler={() => navigation.push('QR Scanner')}  />
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  item: {
    paddingTop: 12,
    paddingBottom: 12,
    paddingLeft: 24,
  },
  name: {
    fontSize: 20,
    paddingBottom: 12,

  },
  index: {
    textAlign: 'right',
    paddingRight: 24,
    fontSize: 20,
    color: 'grey',
  },
  type: {
    paddingLeft: 24,
  },
  itemBreaker: {
    height: 1,
    backgroundColor: 'grey',
    alignSelf: 'center',
  },
})