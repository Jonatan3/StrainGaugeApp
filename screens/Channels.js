import React from 'react'
import { StyleSheet, View, Text, FlatList, TouchableWithoutFeedback } from 'react-native'
import StreamingConstructor from '../constructors/StreamingConstructor';
import QRCodeButtonNoText from '../components/QRCodeButtonNoText';
import colors from '../resources/colors';
import { Dimensions } from 'react-native'
import StreamingConnector from '../constructors/StreamingConnector';

export default function Channels({ navigation }) {

  const streamingConnector = new StreamingConnector()
  streamingConnector.connect()

  const windowWidth = Dimensions.get('window').width;
  const windowHeight = Dimensions.get('window').height;

  return (
    <View style={{ flex: 1 }}>
      <FlatList
        data={StreamingConstructor.getInstance().getDummyData()}
        style={{ height: '100%' }}
        keyExtractor={(item, index) => item.id}
        renderItem={({ item, index }) => (
          <TouchableWithoutFeedback onPress={() => navigation.push('Strain Gauge', {itemId: item.id})}>
            <View>
              <View style={[styles.item, {backgroundColor: index % 2 == 0  ? "white" : colors.channelGrey }]}>
                <View style={{ flexDirection: 'row' }}>
                  <View style={{ flex: 1 }}>
                    <Text style={styles.name}> {item.name} </Text>
                  </View>
                  <View style={{ flex: 1 }}>
                    <Text style={styles.index}> {index < 9 ? '0' + (index + 1) : (index + 1) } </Text>
                  </View>
                </View>
                <Text> {item.type} </Text>
              </View>
            </View>
          </TouchableWithoutFeedback>
        )}
      />
      {/* TODO change this back to : <View style={{ flex: 1, alignSelf: 'flex-end', justifyContent: 'flex-end', paddingBottom: 30, paddingRight: 30 }}>
        <QRCodeButtonNoText onPressHandler={() => navigation.push('QR Scanner')}  />
      </View> 
      MEN DET ER LORT!!!!*/}
      <View style={{ position: 'absolute', top: (windowHeight/1.25), right: (windowWidth / 10) }}>
        <QRCodeButtonNoText onPressHandler={() => navigation.push('QR Scanner')}  />
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  item: {
    paddingTop: 12,
    paddingBottom: 12,
    paddingLeft: 24,
  },
  name: {
    fontSize: 20,
    paddingBottom: 12,

  },
  index: {
    textAlign: 'right',
    paddingRight: 24,
    fontSize: 20,
    color: 'grey',
  },
  type: {
    paddingLeft: 24,
  },
  itemBreaker: {
    height: 1,
    backgroundColor: 'grey',
    alignSelf: 'center',
  },
})