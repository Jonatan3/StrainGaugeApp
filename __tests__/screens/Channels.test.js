import React from 'react';
import renderer from 'react-test-renderer';
import Channels from '../../screens/Channels'

test('renders correctly', () => {
    const tree = renderer.create(<Channels />).toJSON()
    expect(tree).toMatchSnapshot()
})