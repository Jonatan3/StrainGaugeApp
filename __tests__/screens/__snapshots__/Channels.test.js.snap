// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        DummyChannel {
          "channelData": Array [
            Object {
              "x": 0,
              "y": 0,
            },
          ],
          "connectionStatus": undefined,
          "generateDataPoint": [Function],
          "getChannelData": [Function],
          "getConnectionStatus": [Function],
          "getId": [Function],
          "getLatestDataPoint": [Function],
          "getName": [Function],
          "getType": [Function],
          "id": 5156,
          "interval": null,
          "name": "Goddag",
          "type": "Half bridge",
        },
        DummyChannel {
          "channelData": Array [
            Object {
              "x": 0,
              "y": 0,
            },
          ],
          "connectionStatus": true,
          "generateDataPoint": [Function],
          "getChannelData": [Function],
          "getConnectionStatus": [Function],
          "getId": [Function],
          "getLatestDataPoint": [Function],
          "getName": [Function],
          "getType": [Function],
          "id": 8739,
          "interval": null,
          "name": "Goddag",
          "type": "Half bridge",
        },
        DummyChannel {
          "channelData": Array [
            Object {
              "x": 0,
              "y": 0,
            },
          ],
          "connectionStatus": true,
          "generateDataPoint": [Function],
          "getChannelData": [Function],
          "getConnectionStatus": [Function],
          "getId": [Function],
          "getLatestDataPoint": [Function],
          "getName": [Function],
          "getType": [Function],
          "id": 3959,
          "interval": null,
          "name": "Goddag",
          "type": "Half bridge",
        },
        DummyChannel {
          "channelData": Array [
            Object {
              "x": 0,
              "y": 0,
            },
          ],
          "connectionStatus": true,
          "generateDataPoint": [Function],
          "getChannelData": [Function],
          "getConnectionStatus": [Function],
          "getId": [Function],
          "getLatestDataPoint": [Function],
          "getName": [Function],
          "getType": [Function],
          "id": 6901,
          "interval": null,
          "name": "Goddag",
          "type": "Half bridge",
        },
        DummyChannel {
          "channelData": Array [
            Object {
              "x": 0,
              "y": 0,
            },
          ],
          "connectionStatus": true,
          "generateDataPoint": [Function],
          "getChannelData": [Function],
          "getConnectionStatus": [Function],
          "getId": [Function],
          "getLatestDataPoint": [Function],
          "getName": [Function],
          "getType": [Function],
          "id": 56,
          "interval": null,
          "name": "Goddag",
          "type": "Half bridge",
        },
        DummyChannel {
          "channelData": Array [
            Object {
              "x": 0,
              "y": 0,
            },
          ],
          "connectionStatus": true,
          "generateDataPoint": [Function],
          "getChannelData": [Function],
          "getConnectionStatus": [Function],
          "getId": [Function],
          "getLatestDataPoint": [Function],
          "getName": [Function],
          "getType": [Function],
          "id": 489,
          "interval": null,
          "name": "Goddag",
          "type": "Half bridge",
        },
        DummyChannel {
          "channelData": Array [
            Object {
              "x": 0,
              "y": 0,
            },
          ],
          "connectionStatus": true,
          "generateDataPoint": [Function],
          "getChannelData": [Function],
          "getConnectionStatus": [Function],
          "getId": [Function],
          "getLatestDataPoint": [Function],
          "getName": [Function],
          "getType": [Function],
          "id": 4079,
          "interval": null,
          "name": "Goddag",
          "type": "Half bridge",
        },
        DummyChannel {
          "channelData": Array [
            Object {
              "x": 0,
              "y": 0,
            },
          ],
          "connectionStatus": true,
          "generateDataPoint": [Function],
          "getChannelData": [Function],
          "getConnectionStatus": [Function],
          "getId": [Function],
          "getLatestDataPoint": [Function],
          "getName": [Function],
          "getType": [Function],
          "id": 5579,
          "interval": null,
          "name": "Goddag",
          "type": "Half bridge",
        },
        DummyChannel {
          "channelData": Array [
            Object {
              "x": 0,
              "y": 0,
            },
          ],
          "connectionStatus": true,
          "generateDataPoint": [Function],
          "getChannelData": [Function],
          "getConnectionStatus": [Function],
          "getId": [Function],
          "getLatestDataPoint": [Function],
          "getName": [Function],
          "getType": [Function],
          "id": 4419,
          "interval": null,
          "name": "Goddag",
          "type": "Half bridge",
        },
        DummyChannel {
          "channelData": Array [
            Object {
              "x": 0,
              "y": 0,
            },
          ],
          "connectionStatus": false,
          "generateDataPoint": [Function],
          "getChannelData": [Function],
          "getConnectionStatus": [Function],
          "getId": [Function],
          "getLatestDataPoint": [Function],
          "getName": [Function],
          "getType": [Function],
          "id": 7011,
          "interval": null,
          "name": "Goddag",
          "type": "Half bridge",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "height": "100%",
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 12,
                  "paddingLeft": 24,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "white",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 20,
                      "paddingBottom": 12,
                    }
                  }
                >
                   
                  Goddag
                   
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "grey",
                      "fontSize": 20,
                      "paddingRight": 24,
                      "textAlign": "right",
                    }
                  }
                >
                   
                  01
                   
                </Text>
              </View>
            </View>
            <Text>
               
              Half bridge
               
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 12,
                  "paddingLeft": 24,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "#F8F8FD",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 20,
                      "paddingBottom": 12,
                    }
                  }
                >
                   
                  Goddag
                   
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "grey",
                      "fontSize": 20,
                      "paddingRight": 24,
                      "textAlign": "right",
                    }
                  }
                >
                   
                  02
                   
                </Text>
              </View>
            </View>
            <Text>
               
              Half bridge
               
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 12,
                  "paddingLeft": 24,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "white",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 20,
                      "paddingBottom": 12,
                    }
                  }
                >
                   
                  Goddag
                   
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "grey",
                      "fontSize": 20,
                      "paddingRight": 24,
                      "textAlign": "right",
                    }
                  }
                >
                   
                  03
                   
                </Text>
              </View>
            </View>
            <Text>
               
              Half bridge
               
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 12,
                  "paddingLeft": 24,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "#F8F8FD",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 20,
                      "paddingBottom": 12,
                    }
                  }
                >
                   
                  Goddag
                   
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "grey",
                      "fontSize": 20,
                      "paddingRight": 24,
                      "textAlign": "right",
                    }
                  }
                >
                   
                  04
                   
                </Text>
              </View>
            </View>
            <Text>
               
              Half bridge
               
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 12,
                  "paddingLeft": 24,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "white",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 20,
                      "paddingBottom": 12,
                    }
                  }
                >
                   
                  Goddag
                   
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "grey",
                      "fontSize": 20,
                      "paddingRight": 24,
                      "textAlign": "right",
                    }
                  }
                >
                   
                  05
                   
                </Text>
              </View>
            </View>
            <Text>
               
              Half bridge
               
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 12,
                  "paddingLeft": 24,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "#F8F8FD",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 20,
                      "paddingBottom": 12,
                    }
                  }
                >
                   
                  Goddag
                   
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "grey",
                      "fontSize": 20,
                      "paddingRight": 24,
                      "textAlign": "right",
                    }
                  }
                >
                   
                  06
                   
                </Text>
              </View>
            </View>
            <Text>
               
              Half bridge
               
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 12,
                  "paddingLeft": 24,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "white",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 20,
                      "paddingBottom": 12,
                    }
                  }
                >
                   
                  Goddag
                   
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "grey",
                      "fontSize": 20,
                      "paddingRight": 24,
                      "textAlign": "right",
                    }
                  }
                >
                   
                  07
                   
                </Text>
              </View>
            </View>
            <Text>
               
              Half bridge
               
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 12,
                  "paddingLeft": 24,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "#F8F8FD",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 20,
                      "paddingBottom": 12,
                    }
                  }
                >
                   
                  Goddag
                   
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "grey",
                      "fontSize": 20,
                      "paddingRight": 24,
                      "textAlign": "right",
                    }
                  }
                >
                   
                  08
                   
                </Text>
              </View>
            </View>
            <Text>
               
              Half bridge
               
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 12,
                  "paddingLeft": 24,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "white",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 20,
                      "paddingBottom": 12,
                    }
                  }
                >
                   
                  Goddag
                   
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "grey",
                      "fontSize": 20,
                      "paddingRight": 24,
                      "textAlign": "right",
                    }
                  }
                >
                   
                  09
                   
                </Text>
              </View>
            </View>
            <Text>
               
              Half bridge
               
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 12,
                  "paddingLeft": 24,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "#F8F8FD",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 20,
                      "paddingBottom": 12,
                    }
                  }
                >
                   
                  Goddag
                   
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "grey",
                      "fontSize": 20,
                      "paddingRight": 24,
                      "textAlign": "right",
                    }
                  }
                >
                   
                  10
                   
                </Text>
              </View>
            </View>
            <Text>
               
              Half bridge
               
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "position": "absolute",
        "right": 75,
        "top": 1067.2,
      }
    }
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2DA39D",
              "borderBottomWidth": 2,
              "borderColor": "#00817C",
              "borderRadius": 100,
              "borderRightWidth": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "margin": 10,
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="36px"
              bbWidth="36px"
              enable-background="new 0 0 24 24"
              fill="#FFFFFF"
              focusable={false}
              height="36px"
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 36,
                    "width": 36,
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
              width="36px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <RNSVGGroup
                fill={4294967295}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGRect
                  fill={null}
                  height="24"
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  width="24"
                  x={0}
                  y={0}
                />
                <RNSVGPath
                  d="M9.5,6.5v3h-3v-3H9.5 M11,5H5v6h6V5L11,5z M9.5,14.5v3h-3v-3H9.5 M11,13H5v6h6V13L11,13z M17.5,6.5v3h-3v-3H17.5 M19,5h-6v6 h6V5L19,5z M13,13h1.5v1.5H13V13z M14.5,14.5H16V16h-1.5V14.5z M16,13h1.5v1.5H16V13z M13,16h1.5v1.5H13V16z M14.5,17.5H16V19h-1.5 V17.5z M16,16h1.5v1.5H16V16z M17.5,14.5H19V16h-1.5V14.5z M17.5,17.5H19V19h-1.5V17.5z M22,7h-2V4h-3V2h5V7z M22,22v-5h-2v3h-3v2 H22z M2,22h5v-2H4v-3H2V22z M2,2v5h2V4h3V2H2z"
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
